CMSgov/mdct-mcr
# focus errored fields on blocked form submission

## Description

This is a proof of concept for how we could focus errored fields on blocked form submission.

Changes:
- Field `id` and `name` needed to have a differentiator (in this case, `abf` for 'admin banner form' added so that elements could be correctly selected. (There were collisions with `title, and `description`, and this kind of pattern guards us from similar sneaky bugs in the future.) This also required some changes to other files to make sure we were being consistent.
- Had to add an explicit ref & register call to `TextField` so the order of errors could follow the field display order (i.e. if there are multiple co-occurring errors in the form, the first error is the one that gets focus).
- Added two new utility methods to handle 

Note: `react-hook-form` is supposed to be able to handle focusing fields on error, but they note in their documentation that this is sometimes prohibitively difficult to configure with 3rd-party design system components, and suggest disabling it if that is the case. After trying to get their built-in functionality to work, I've settled on making our own as shown in this PR.

### How to test

1. Log in as admin and nav to /admin.
2. Attempt form submission with invalid form fill. The first field with an error should become focused, and if necessary, window will scroll to that position.

### Changed Dependencies

n/a

## Code author checklist

## Reviewer checklist (two different people)
